packages/nodes-base/nodes/Hubspot/GenericFunctions.ts

```diff
@@ -38,16 +38,8 @@ export async function hubspotApiRequest(this: IHookFunctions | IExecuteFunctions
 	};
 
 	try {
-		if (authenticationMethod === 'apiKey') {
-			const credentials = await this.getCredentials('hubspotApi');
-
-			options.qs.hapikey = credentials.apiKey as string;
-			return await this.helpers.request!(options);
-		} else if (authenticationMethod === 'appToken') {
-			const credentials = await this.getCredentials('hubspotAppToken');
-
-			options.headers!['Authorization'] = `Bearer ${credentials.appToken}`;
-			return await this.helpers.request!(options);
+		if (authenticationMethod === 'apiKey' || authenticationMethod === 'appToken') {
+			return this.helpers.requestWithAuthentication.call(this, authenticationMethod, options);
```

Comment:
Hey @vcrwr,

Just testing and I realised I messed up... This needs to be...

```
const credentialType = authenticationMethod === 'apiKey' ? 'hubspotApi' : 'hubspotAppToken';
return this.helpers.requestWithAuthentication.call(this, credentialType, options);
```

I forgot that the Hubspot node is one of the few where the authentication method doesn't match the actual credential name.

If you can get this part added in the rest of it is good and I can get it merged.

Reply:
Thats weird. I saw this and then decided to test it manually too. I was able to pull data from the API with both credentialTypes. Let me check, what I did wrong on the testing side.

Reply:
In the moment I wrote it, I see the issue. I tested with the HttpNode. ;)

Reply:
While thinking about it, I'd rather would switch back to the original variant. The need for another variable is symbolic for the increase of complexity. Its less readable in the new version IMHO. Of course the best way would be, to adjust the Hubspot Credential Naming, but as this breaks with existing installations, this is surely not an option.

@Joffcom what do you think about rolling back to the original if else if variant?

Reply:
@vcrwr I like the smaller version as it reduces reused code. But both ways would work.

Reply:
Ok. Happy to apply your smaller version then. I'll give it a bit more testing tomorrow, before committing.

Reply:
I will give it another test as well just to be safe. Thanks for going through this.

Reply:
smaller version committed and tested successfully.